provider/image.go

```diff
@@ -175,15 +175,37 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 	if err != nil {
 		return "", nil, err
 	}
-	// Quick and dirty auth; we can also preconfigure the client itself I believe
-	// TODO: use auth pattern as above to use default auth
-	var authConfig = types.AuthConfig{
-		Username:      img.Registry.Username,
-		Password:      img.Registry.Password,
-		ServerAddress: img.Registry.Server,
+
+	// authentication for registry push
+	// we check if the user set creds in the Pulumi program, and use those preferentially
+	var pushAuthConfig types.AuthConfig
+
+	if img.Registry.Username != "" && img.Registry.Password != "" {
+		pushAuthConfig.Username = img.Registry.Username
+		pushAuthConfig.Password = img.Registry.Password
+		pushAuthConfig.ServerAddress = img.Registry.Server
+	} else {
+		// send warning if user is attempting to use in-program credentials
+		if img.Registry.Username == "" && img.Registry.Password != "" {
+			msg := "username was not set, although password was; using host credentials file"
+			err = p.host.LogStatus(ctx, "warning", urn, msg)
```

Comment:
My understanding is that `LogStatus` is a transient message, but I think a permanent message would be appropriate here. We want the warning to be discoverable by the user after the program has been run.
```suggestion
			err = p.host.Log(ctx, "warning", urn, msg)
```